modules/xmpp/SdpConsistency.js

```diff
@@ -0,0 +1,139 @@
+var transform = require('sdp-transform');
+
+/**
+ * Begin helper functions
+ */
+function getPrimarySsrc (videoMLine) {
+    if (!videoMLine.ssrcs) {
+        return;
+    }
+    let numSsrcs = videoMLine.ssrcs
+        .filter(ssrcInfo => ssrcInfo.attribute === "msid")
+        .map(ssrcInfo => ssrcInfo.id)
+        .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
+        .length;
+    if (numSsrcs === 1) {
+        return videoMLine.ssrcs
+            .filter(ssrcInfo => ssrcInfo.attribute === "msid")
+            .map(ssrcInfo => ssrcInfo.id)[0];
+    } else {
+        let findGroup = (mLine, groupName) => {
+            return mLine
+                .ssrcGroups
+                .filter(group => group.semantics === groupName)[0];
+        };
+        // Look for a SIM or FID group
+        if (videoMLine.ssrcGroups) {
+            let simGroup = findGroup(videoMLine, "SIM");
+            if (simGroup) {
+                return parseInt(simGroup.ssrcs.split(" ")[0]);
+            }
+            let fidGroup = findGroup(videoMLine, "FID");
+            if (fidGroup) {
+                return parseInt(fidGroup.ssrcs.split(" ")[0]);
+            }
+        }
+    }
+}
+
+function getRtxSsrc (videoMLine, primarySsrc) {
+    if (videoMLine.ssrcGroups) {
+        return videoMLine
+            .ssrcGroups
+            .filter(ssrcGroup => ssrcGroup.semantics === "FID")
+            .filter(ssrcGroup => 
+                parseInt(ssrcGroup.ssrcs.split(" ")[0]) === primarySsrc)
+            .map(ssrcGroup => ssrcGroup.ssrcs.split(" ")[1])
+            .map(ssrcStr => parseInt(ssrcStr))[0];
+    }
+}
+/**
+ * End helper functions
+ */
+
+/**
+ * Handles the work of keeping video ssrcs consistent across multiple
+ * o/a cycles, making it such that all stream operations can be
+ * kept local and do not need to be signaled.
+ * NOTE: This only keeps the 'primary' video ssrcs consistent: meaning
+ * the primary video stream and an associated RTX stream, if it exists
+ */
+class SdpConsistency {
+    constructor () {
```

Comment:
i already have a js doc above the class, is there something specific in there that i'm missing?